aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/analytics/AnalyticsClient.java

```diff
@@ -359,33 +359,38 @@ public void removeGlobalMetric(String eventType, String metricName) {
     }
 
     /**
-     * Adds the specified campaign attributes to events to track Campaign Analytic
-     * <p>
-     * You should not use this method as it will be called by the NotificationManager when the app is opened
-     * from a push notification.
-     *
-     * @param campaign the map with campaign attributes of the campaign received
+     * Replaces the current event source attributes (if any)
+     * with a new set of event source attributes
+     * @param attributes map of attribute values
      */
-    public void setCampaignAttributes(Map<String, String> campaign) {
-        if (campaign == null) {
-            log.warn("Null campaign attributes provided to setCampaignAttributes.");
-            return;
+    public void updateEventSourceGlobally(final Map<String, String> attributes) {
```

Comment:
Where is this being called from? It's part of the public API, no? If anyone has `pinpointManager.getAnalyticsClient().setCampaignAttributes(...)` in their code base, this would be a breaking change.

Like for the other method, you could _add_ this method, and have it pass to an `@Deprecated public void setCampaignAttributes(....)`.

Reply:
Good call on this one. I'll add this back in and deprecate it.

Reply:
Arg, actually the existing method had:
```
You should not use this method as it will be called by the NotificationManager when the app is opened
     * from a push notification.
```
Why was this method `public` originally? A customer isn't supposed to actually call it I guess?